File: lab2/task_1/src/Circle.java
import java.util.Scanner;

public class Circle {
    private double radius;
    private double x;
    private double y;

    public void init(double r, double arg_x, double arg_y) {
        radius = r;
        x = arg_x;
        y = arg_y;
    }

    public void read() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Введите радиус: ");
        radius = sc.nextDouble();
        System.out.print("Введите координату x: ");
        x = sc.nextDouble();
        System.out.print("Введите координату y: ");
        y = sc.nextDouble();
    }

    public void display() {
        System.out.println("Радиус: " + radius);
        System.out.println("Координата x: " + x);
        System.out.println("Координата y: " + y);
    }

    public double distance() {
        return Math.sqrt(x * x + y * y);
    }

    public Circle add(Circle c1, Circle c2) {
        Circle result = new Circle();
        result.init(c1.radius + c2.radius, c1.x + c2.x, c1.y + c2.y);
        return result;
    }
}

File: lab2/task_3/Lab2/Circle.cs
using System;

public class Circle
{
    private double radius;
    private double x;
    private double y;

    public double Radius {
      get { return radius; }
      set { radius = value; }
    }

    public void Init(double r, double arg_x, double arg_y)
    {
        Radius = r;
        x = arg_x;
        y = arg_y;
    }

    public void Read()
    {
        Console.Write("Введите радиус: ");
        radius = Convert.ToDouble(Console.ReadLine());
        Console.Write("Введите координату x: ");
        x = Convert.ToDouble(Console.ReadLine());
        Console.Write("Введите координату y: ");
        y = Convert.ToDouble(Console.ReadLine());
    }

    public void Display()
    {
        Console.WriteLine($"Радиус: {radius}");
        Console.WriteLine($"Координата x: {x}");
        Console.WriteLine($"Координата y: {y}");
    }

    public double Distance()
    {
        return Math.Sqrt(x * x + y * y);
    }

    public Circle Add(Circle c)
    {
        Circle result = new Circle();
        result.Init(radius + c.radius, x + c.x, y + c.y);
        return result;
    }
}
